### PR TITLE
[FEAT] Add featured themes to home

### PR DESCRIPTION
**Pull request by kyjus25**

### CODE REVIEW

**Comment by CatPerson136** [comment]
Can you explain what you did?

**Comment by kyjus25** [comment]
@CatPerson136 Of course! The backend exposes some featured themes at https://api.pulsar-edit.dev/api/themes/featured in addition to the regular list of featured packages that we were already pulling. I am just pulling in those also at the bottom of the page

featured themes

**Comment by CatPerson136** [comment]
That is awesome, I will love to see it added to pulsar. Only if I could merge it myself, I would.

File: src/handlers.js [hunk 1]
```diff
@@ -110,20 +110,28 @@ async function homePage(req, res, timecop) {
...
+      ]).then(([featuredPackages, featuredThemes]) => Promise.all([
+        utils.prepareForListing(featuredPackages.body),
+        utils.prepareForListing(featuredThemes.body)
+      ])).then(([featuredPackages, featuredThemes]) => ({
+        featuredPackages,
+        featuredThemes
+      }));
       timecop.end("api-request");
       timecop.start("transcribe-json");
-      let obj = await utils.prepareForListing(api.body);
```
Comment (CatPerson136) [inline src/handlers.js:122]: Why was this removed?

**Review by CatPerson136 — APPROVED** [review]
Looks good, still confused why line 122 was removed though? But I think approve of these changes.

File: src/handlers.js (hunk 1 above)
Comment (kyjus25) [reply src/handlers.js:122]: @CatPerson136 I appear to not be able to copy line references on the iPhone app, but it was deleted and moved to line 123

Comment (kyjus25) [reply src/handlers.js:122]: Updated!

File: src/handlers.js [hunk 2]
```diff
@@ -110,18 +110,31 @@ async function homePage(req, res, timecop) {
   if (cached !== null) {
     timecop.end("cache-check");
     // We know our cache is good and lets serve the data
-    res.render("home", { featured: cached, timecop: timecop.timetable, page: homePage });
+    res.render("home", { ...cached, timecop: timecop.timetable, page: homePage });
```
Comment (confused-Techie) [inline src/handlers.js:113]: I'll also want to be aboslutely sure that the cache still functions with both options. While it should, just because it's written as JSON, I'd want to make sure the two objects are parsed as JSON text properly

**Review by confused-Techie — COMMENTED** [review]
Just a few thoughts on here, that I want to see what you think. Otherwise I'll try to test this all out soon enough.

File: src/handlers.js (hunk 2 above)
Comment (kyjus25) [reply src/handlers.js:113]: Sounds good. Yeah, theoretically it should work, but I am not familiar with Google storage to test. There may be a hiccup on the first initialization where the cache does not match the new `{featuredPackages: [], featuredThemes: []}` object.

File: src/handlers.js [hunk 3]
```diff
@@ -110,18 +110,31 @@ async function homePage(req, res, timecop) {
...
+        superagent.get(`${apiurl}/api/packages/featured`),
+        superagent.get(`${apiurl}/api/themes/featured`)
+      ])
+      .then(obj => {
+        timecop.end("api-request");
+        timecop.start("transcribe-json")
+        return obj;
+      })
+      .then(([featuredPackages, featuredThemes]) => Promise.all([
+        utils.prepareForListing(featuredPackages.body),
```
[the remark this answers is not in the document]
Comment (kyjus25) [reply src/handlers.js:129]: @confused-Techie If I understand what you're asking, yes, when you return something in a `.then()` it affects the end output; it allows you to chain promises. Definitely feel free to check it, but from my testing it works as it did before, just brings in the featured themes to the bottom of the page also

File: src/handlers.js (hunk 2 above)
Comment (confused-Techie) [reply src/handlers.js:113]: Yeah good point. There shouldn't be, but after deployment I'll go ahead and manually delete the cache just in case.

**Comment by confused-Techie** [comment]
Also @kyjus25 while this looks good there is currently a conflict after merged your previous changes

**Comment by kyjus25** [comment]
No worries, I will address. Thanks!

**Comment by kyjus25** [comment]
@confused-Techie This has been updated

**Review by confused-Techie — APPROVED** [review]
